pkg/storage/unified/sql/backend.go

```diff
@@ -373,6 +452,22 @@ func (b *backend) ReadResource(ctx context.Context, req *resource.ReadRequest) *
 		res, err = dbutil.QueryRow(ctx, tx, sr, readReq)
 		return err
 	})
+
+	// if not found, look for latest deleted version (if requested)
```

Comment:
NIT – I think we could / should do that in the same TX as the normal read, for consistency.

Reply:
Good idea - updated